### ISSUE TITLE
MiniZinc Challenge 2016 - error: array index mismatch

### ISSUE THREAD

**Issue by PatrickTrentin88** [MiniZinc/minizinc.github.io#20]
The new `MiniZinc` version `2.3.1+` is not compatible with one of the benchmarks used in the *MiniZinc Challenge 2016*. It prints the following error:

```
MiniZinc: flattening error: 
  /.../cryptanalysis/step1_aes.mzn:44:
  in call 'SR'
  array index set 1 of argument 2 does not match declared index set
```

Looking at the model, it appears that the newer compiler found an issue with the underlying model which may or may not affect the correctness of the results computed by all FlatZinc solvers on this benchmark.

Version `2.2.1` runs fine.

The model causing this issue is [step1_aes.mzn](https://github.com/MiniZinc/minizinc.github.io/blob/master/challenge2016/mznc2016_probs/cryptanalysis/step1_aes.mzn), with any data-set.

**Comment by Dekker1** [comment]
Hi Patrick,

We had noticed the problem. The model has been fixed in the MiniZinc benchmarks repostiroy: https://github.com/MiniZinc/minizinc-benchmarks/commit/26bcd0a78433025f7b6896a6fa8caf128795760b. 

In the model there was a mistake in the declaration of the domain of the arguments. Even though that is incorrect, previous versions of MiniZinc did not check these. In the newer versions this causes an error.

Even though the model was not fully correct, it doesn't invalidate the MiniZinc challenge results. The FlatZinc generated from the old model was correct as the body of the predicate did not change.
